### DISCUSSION TITLE
Optimise Basic ASIO4All set-up

### DISCUSSION THREAD

**Discussion by TPennick** [jamulussoftware/jamulus#1300] [Technical Support & Help]
Hi All,
Hoping to start a regular session for a group of folk musicians who haven't met for over a year, I suggested Jamulus.  Two of the group have set up a basic configuration with ASIO4All, and mic and headphones plugged directly into their Windows laptops.  However, in both cases they are experiencing unacceptable echo when connected to my private server at a Ping distance of about 6ms.  I don't use ASIO4All myself as I have a Behringer UMC device.  I'm still climbing the foot-hills of the Jamulus learning curve.  I'm also blind, so visiting them to attempt to sort out their issue would have been tricky at the best of times.  Can anyone suggest easy steps that I can pass on to them to get the best out of their set-up, though I realise the best option would be to buy a device with an ASIO driver, I don't want to suggest to them that they shell out cash for something they haven't tried yet.
Thanks for any suggestions,
Tim Pennick

**Comment by hoffie** [comment]
Hi,

can you describe what kind of echo they are experiencing?

Possible reasons:
- There's a reverb in Jamulus which is often confused with some gain control. This might sound like an echo.
- Sometimes people forget to use headphones or don't realize that their PC speakers are active. This can usually be solved by changing the ASIO4ALL configuration.
- Another common ASIO4ALL misconfiguration is using something like "Stereo input" or "Stereo mix" as input. This is usually the signal which is intended to go to your headphones. This creates problems, although usually in the form of feedback.
- Could it be that they are just hearing themselves from the server signal? To check, they can mute themselves in their mix. However, it is usually not recommended to do that while playing/singing as one should always be listening to the signal from the server (even if it is confusing at first).

Kind regards,
Christian

**Comment by TPennick** [comment]
Hi Christian,

 

Very many thanks for your reply.  I realise this is all a bit vague and appreciate your suggestions.  What they were hearing was delay rather than reverb.  They had both managed to prevent sound being produced by the speakers of their machines.  It may well be that they had achieved the best results they could get with ASIO4ALL, but as a non-ASIO4ALL user, I was wondering if there are any further tips and tricks for reducing the latency in the system, and also I guess, what sort of results they can expect with this type of set-up.

 

Regards and thanks again,

 

Tim

**Reply by hoffie** [reply]
> What they were hearing was delay rather than reverb.

Ok, then it was probably just their own signal with the delay from the client-to-server-to-client roundtrip.

> It may well be that they had achieved the best results they could get with ASIO4ALL, but as a non-ASIO4ALL user, I was wondering if there are any further tips and tricks for reducing the latency in the system, and also I guess, what sort of results they can expect with this type of set-up.

I don't think ASIO4ALL is bad per-se. Configuration is sometimes hard to get right and it's often used to enable usage of less-than-optimal sound equipment. :)

Common reasons for high delay include:
- Sound capture delays. You can try to reduce them by setting the lowest working capture buffer (64) on the left of the settings menu. If this works on that machine, you can and should also tick the "Use small network buffers" checkbox.
- Network delays. Choose a server with small ping time. Don't use Wifi. Avoid any concurrent downloads/uploads in the same network. Manually reduce Jitter buffer size in Jamulus settings (although the Auto algorithm usually does a good job and tends to optimize for low latency rather than zero dropouts, in my experience).
- There may be additional sound capture and playback delays, especially when using bluetooth headsets. Try using cables instead. :)

In the end, some delay has to be accepted. One gets used to it. :)
Good headphones (which remove all sound from the environment) help.

Kind regards,
Christian
